src/crawlee/basic_crawler/_basic_crawler.py

```diff
@@ -1061,3 +1062,15 @@ async def __run_task_function(self) -> None:
 
     async def __run_request_handler(self, context: BasicCrawlingContext) -> None:
         await self._context_pipeline(context, self.router)
+
+
+@docs_group('Classes')
+@dataclass(frozen=True)
+class BlockedInfo:
+    """Information about whether the crawling is blocked. If reason is empty, then it means it is not blocked."""
+
+    reason: str
+
+    def __bool__(self) -> bool:
+        """No reason means no blocking."""
+        return bool(self.reason)
```

Comment:
The `BlockedInfo` should be included in the "Data Structures" group. 

I also find problematic that it is buried around line one thousand, following an really long class definition. Please move it either to the top of the file or to a separate file.

Reply:
Done